### DISCUSSION TITLE
PrimeFaces 11, InputNumber IE 11 Compatibility

### DISCUSSION THREAD

**Discussion by RobiVeho** [primefaces/community#76] [PrimeFaces] [answered]
In one of our projects we are using PF 11. Unfortunately, we have many users who were still using IE 11. According to the documentation, Primefaces should be compatible with IE 11.

We noticed a problem using the InputNumber component in IE11. It seems that there is an issue when converting NodeList to Array. In InputNumber.js we can find the following code:

  ```
 "./src/AutoNumericHelper.js":
         /*!****************************************!*\
     !*** ./src/AutoNumericHelper.js ***!
     \************************************/
         /*! exports provided: default */
         function (c, i, e) {
             e.r(i);
             e.d(i, "default", function () {
                 return t;
             });
             var a = e("./src/AutoNumericEnum.js");
             function g(u) {
                 return o(u) || r(u) || k(u) || b();
             }
         
             function b() {
                 throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
             }
             function r(u) {
                 if (typeof Symbol !== "undefined" && Symbol.iterator in Object(u)) {
                     return Array.from(u);
                 }
             }
             function o(u) {
                 if (Array.isArray(u)) {
                     return s(u);
                 }
             }
```

In  IE 11, the b method is always triggered, since o(u), r(u) and k(u) can't convert Nodelist into Array. 
We suggest the following fix:

```


   var a = e("./src/AutoNumericEnum.js");
             function g(u) {
                 return o(u) || r(u) || k(u) || ie11(u) || b();
             }
             function ie11(u) {
                 return Array.prototype.slice.call(u);
             }
             function b() {
                 throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
             }
```

Is it possible to expect that this issue will be fixed in one of the next versions? And generally how long can we expect compatibilty with IE 11?

Robert

**Comment by FlipWarthog** [comment, marked as answer]
IE support has already been dropped in PrimeFaces 13. See this ticket for additional information: https://github.com/primefaces/primefaces/issues/6956

**Reply by FlipWarthog** [reply]
IE support was also dropped in autoNumeric, the lib used by InputNumber, as well. https://github.com/autoNumeric/autoNumeric/issues/764

**Reply by jepsar** [reply]
Also, IE 11 is no longer supported by MS for over a year already https://learn.microsoft.com/en-us/lifecycle/faq/internet-explorer-microsoft-edge#is-internet-explorer-11-the-last-version-of-internet-explorer-
